cmd/format/format.go

```diff
@@ -173,11 +174,27 @@ func Run(v *viper.Viper, statz *stats.Stats, cmd *cobra.Command, paths []string)
 			return fmt.Errorf("exactly one path should be specified when using the --stdin flag")
 		}
 	} else {
-		// checks all paths are contained within the tree root
-		for _, path := range paths {
-			rootPath := filepath.Join(cfg.TreeRoot, path)
-			if _, err = os.Stat(rootPath); err != nil {
-				return fmt.Errorf("path %s not found within the tree root %s", path, cfg.TreeRoot)
+		// checks all paths are contained within the tree root and exist
+		// also "normalize" paths so they're relative to cfg.TreeRoot
+		for i, path := range paths {
+			absolutePath, err := filepath.Abs(path)
+			if err != nil {
+				return fmt.Errorf("error computing absolute path of %s: %w", path, err)
+			}
+
+			relativePath, err := filepath.Rel(cfg.TreeRoot, absolutePath)
+			if err != nil {
+				return fmt.Errorf("error computing relative path from %s to %s: %s", cfg.TreeRoot, absolutePath, err)
+			}
+
+			if strings.HasPrefix(relativePath, "..") {
```

Comment:
I honestly don't know if there are issues with this check. I know it would fail to detect paths that start with `/`, but I think that can't happen right after we've called `filepath.Rel`.

This still seems too simple to be right. But we're also just trying to protect people from themselves here, I don't think there's a security issue. It's not like we're not Apache and need to be hyper careful not to serve up files outside of the webroot directory.

Reply:
I agree